src/vm/peimage.inl

```diff
@@ -481,8 +481,19 @@ inline BOOL PEImage::HasNativeHeader()
         return GetLoadedLayout()->HasNativeHeader();
     else
     {
-        PEImageLayoutHolder pLayout(GetLayout(PEImageLayout::LAYOUT_ANY,LAYOUT_CREATEIFNEEDED));
```

Comment:
At least part of the reason was that the `ntHeader.OptionalHeader.Magic` is `IMAGE_NT_OPTIONAL_HDR32_MAGIC` for an IL image, but is expected to be `IMAGE_NT_OPTIONAL_HDR64_MAGIC` for `MAPMapPEFile` on arm64.  https://github.com/dotnet/coreclr/blob/master/src/pal/src/map/map.cpp#L2371-L2378.  When I tried to "fix" this, it failed somewhere else.  It seemed like a bad idea to try to make IL succeed.